src/msix/msix.cpp

```diff
@@ -208,10 +208,37 @@ MSIX_API HRESULT STDMETHODCALLTYPE UnpackBundleFromStream(
     MSIX::ComPtr<IAppxBundleReader> reader;
     ThrowHrIfFailed(factory->CreateBundleReader(stream, &reader));
 
-    auto to = MSIX::ComPtr<IStorageObject>::Make<MSIX::DirectoryObject>(utf8Destination);
+    auto to = MSIX::ComPtr<IDirectoryObject>::Make<MSIX::DirectoryObject>(utf8Destination);
     reader.As<IPackage>()->Unpack(packUnpackOptions, to.Get());
     return static_cast<HRESULT>(MSIX::Error::OK);
 #else
     return static_cast<HRESULT>(MSIX::Error::NotSupported);
 #endif
 } CATCH_RETURN();
+
+#endif // MSIX_UNPACK
+
+#ifdef MSIX_PACK
+
+MSIX_API HRESULT STDMETHODCALLTYPE PackPackage(
+    MSIX_VALIDATION_OPTION validationOption,
+    char* directoryPath,
+    char* outputPackage
+) noexcept try
+{
+    ThrowErrorIfNot(MSIX::Error::InvalidParameter, 
+        (directoryPath != nullptr && outputPackage != nullptr), 
+        "Invalid parameters");
+
+    auto from = MSIX::ComPtr<IDirectoryObject>::Make<MSIX::DirectoryObject>(directoryPath);
+    auto filesMap= from->GetFilesByLastModDate();
+
+    // TODO:
+    // - get stream to manfiest
+    // - add new method to IPackage that takes a std::multimap with the files and stream of the manifest
+
+
+    return static_cast<HRESULT>(MSIX::Error::OK);

```

Comment:
>return static_cast<HRESULT>(MSIX::Error::OK); [](start = 4, length = 45)

NOTIMPLEMENTED

Reply:
I'll do MSIX::Error::NotImplemented

---
In reply to: [276763272](https://github.com/Microsoft/msix-packaging/pull/93#discussion_r276763272) [](ancestors = 276763272)